set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -724,5 +725,58 @@ private void validateAndSetItems(UserSet storedUserSet, UserSet updateUserSet, L
 	    throw new InternalServerException(e);
 	}
     }
+	@DeleteMapping(value = { "/set/delete/" })
+	@ApiOperation(value = "Delete sets associated with user", nickname = "delete user's sets", response = java.lang.Void.class)
+	public ResponseEntity<String> deleteUserAssociatedSet(
+			HttpServletRequest request)
+			throws HttpException {
 
+		// check user credentials, if invalid respond with HTTP 401,
+		// or if unauthorized respond with HTTP 403
+		Authentication authentication = verifyWriteAccess(Operations.DELETE, request);
+
+		return deleteUserAssociatedSets(authentication);
+	}
+
+	/**
+	 * This method implements removal of a all sets associated to a user
+	 *
+	 * @param authentication
+	 * @throws HttpException
+	 */
+	protected ResponseEntity<String> deleteUserAssociatedSets(Authentication authentication) throws HttpException {
+	try {
+		// get the user from authentication
+		Agent user = new WebUser();
+		user.setHttpUrl(getUserSetService().getUserId(authentication));
+		user.setNickname(((ApiCredentials)authentication.getCredentials()).getUserName());
+
+		List<PersistentUserSet> userSets = getUserSetService().getUserSetByCreatorId(user.getHttpUrl());
+
+		// generate “ETag”;
+		String eTagNew = generateETag(userSets.get(0).getModified(), WebFields.FORMAT_JSONLD, getApiVersion());
+
+		// if the user set is disabled and the user is not an admin, respond with HTTP
+		// 410
+		HttpStatus httpStatus = null;
+		httpStatus = HttpStatus.NO_CONTENT;
+
+		// delete user sets
+		getUserSetService().deleteUserSets(userSets);
+
+		// build response entity with headers
+		MultiValueMap<String, String> headers = new LinkedMultiValueMap<>(5);
```

Comment:
The method must return only the following headers

headers.add(HttpHeaders.ALLOW, UserSetHttpHeaders.ALLOW_PGD);
	    headers.add(UserSetHttpHeaders.CACHE_CONTROL, UserSetHttpHeaders.VALUE_NO_CAHCHE_STORE_REVALIDATE);
	    
please create, and use it in DELETE and POST "/set/" methods
UserSetHttpHeaders.ALLOW_PGD = "POST,GET,DELETE";

Reply:
have added ALLOW_PGD = "POST,GET,DELETE"; in UserSetHttpHeaders. have modified the headers